kep/260-network-aware-scheduling/README.md

```diff
@@ -0,0 +1,1511 @@
+			s := &qos.Sort{}
+			return s.Less(pInfo1, pInfo2)
+		}
+
+		// Binary search to find both order index since topology list is ordered by Pod Name
+		var orderP1 = networkAwareUtil.FindPodOrder(appGroup.Status.TopologyOrder, pInfo1.Pod.Name)
+		var orderP2 = networkAwareUtil.FindPodOrder(appGroup.Status.TopologyOrder, pInfo2.Pod.Name)
+
+		klog.Infof("Pod %v order: %v and Pod %v order: %v.", pInfo1.Pod.Name, orderP1, pInfo2.Pod.Name, orderP2)
+
+		// Lower is better, thus invert result!
+		return !(orderP1 > orderP2)
+	} else { // Pods do not belong to the same App Group: follow the strategy from the QoS plugin
+		klog.Infof("Pod %v and %v do not belong to the same appGroup %v", pInfo1.Pod.Name, pInfo2.Pod.Name, p1AppGroup)
+		s := &qos.Sort{}
+		return s.Less(pInfo1, pInfo2)
+	}
+}
+```
+
+#### `TopologicalSort` Example
+
+Let's consider the Online Boutique application shown previously. 
+The AppGroup consists of 11 workloads (11 groups of pods) and the topology order based on the KahnSort algorithm is **P1, P10, P9, P8, P7, P6, P5, P4, P3, P2, P11.**
+
+The plugin favors low indexes. Thus, depending on the two pods evaluated in the Less function, the result (bool) is the following: 
+
+<p align="center"><img src="figs/queueSortExample.png" title="queueSortExample" width="528" class="center"/></p>
+
+
+### Description of the `NetworkOverhead` plugin
+
+#### Extension point: Filter 
+
+Workload dependencies established in the AppGroup CR must be respected.
+
+This plugin focuses on `maxNetworkCost` requirements. 
+
+Nodes providing higher network costs than the `maxNetworkCost` requirement must be filtered out. 
+
+Several dependencies may exist since multiple pods can be already deployed on the network. 
+
+As an initial design, we plan to filter out nodes that unmet a higher number of dependencies to reduce the number of nodes being scored. 
+
+Also, `minBandwidth` requirements will be considered at a later stage. 
+based on the bandwidth capacity / allocatable available in each region / zone. 
+The bandwidth allocatable in each region or zone will be handled by the Network Topology controller previously mentioned.  
+
+```go
+// Filter : evaluate if node can respect maxNetworkCost requirements
+func (pl *NetworkOverhead) Filter(ctx context.Context, cycleState *framework.CycleState, pod *v1.Pod, nodeInfo *framework.NodeInfo) *framework.Status {
+    // Check if Pod belongs to an AppGroup
+    (...)
+    // Check if pods already available, otherwise return
+    (...)
+    // Check Dependencies of the given pod 
+    (...)
+    // Retrieve network costs from the AppGroup CRD based on the region and zone of the node being filtered
+    // Save them in a map to search for costs faster
+    (...)
+    // Main Procedure 
+    var numOK int64 = 0
+    var numNotOK int64 = 0 
+    // check if maxNetworkCost fits 
+    for _, podAllocated := range appGroup.Status.PodsScheduled { // For each pod already allocated 
```

Comment:
This adds burden for the appgroup controller to maintains the scheduled pods; which that info are already covered by other controllers, and can be easily obtained.

Reply:
Already commented above. Thanks for spotting this detail. We updated the description since we are indeed currently using an AppGroup lister for this. Thanks again.